zerver/openapi/zulip.yaml

```diff
@@ -1169,11 +1169,17 @@ paths:
                                 organization administrators will receive this event).
 
                                 This event is also sent when the user gains access to a previously
-                                inaccessible stream either due to user being subscribed to a private
-                                stream or if a private stream is made public.
+                                inaccessible stream, such as when the user is subscribed to a
+                                stream, their role changes, or when a private stream is made public.
```

Comment:
@timabbott do we want to mention about role change in this commit itself? This is being implemented in another PR.

Reply:
I think it would be better to leave the role change part of this sentence out, until those changes are done in the other PR.

I'd revise this to be (with relative links to the help center articles):

> This event is also sent when a user gains access to a stream they previously [could not access](https://zulip.com/help/stream-permissions), such as when a private stream is made public, or when a [guest user](https://zulip.com/help/roles-and-permissions) is subscribed to a public (or private) stream.

It doesn't sound like we're trying to list all the cases that this is can happen here. We just want to give some relevant situations where it's possible. So adding the role change might not be necessary if it's noted in a future feature level **Changes** note.

Reply:
The change you suggested sounds good to me, but this event is also sent when a non-guest user subscribes to a private stream. I know we are trying to phrase suchthat it sounds like we are not listing all the cases, but just wanted to confirm once.

Reply:
Yeah ... what if we used "including but not limited to" instead of "such as"?

> This event is also sent when a user gains access to a stream they previously [could not access](https://zulip.com/help/stream-permissions), including but not limited to when a private stream is made public, or when a [guest user](https://zulip.com/help/roles-and-permissions) is subscribed to a public (or private) stream.

It's a bit formal, but would make it clearer that we're not listing all cases of when a user might gain access to a stream.

Reply:
I think it's fine to do so; a correct client should support it and in my mind it's just a bug, not a conceptual API change, that we're not sending it in that case.